admin/reverseGpx.php

```diff
@@ -43,21 +45,18 @@
 $editfile = $_FILES[$name]['tmp_name'];
 libxml_use_internal_errors(true);
 $adminpg = "../admin/admintools.php";
-$dom = new DOMDocument();
-$dom->formatOutput = true;
-if (!$dom->load($editfile)) {
-    displayGpxUserAlert($gpxfile);
-    header("Location: {$adminpg}");
+validateGpxFile($editfile, $gpxfile);
+if (isset($_SESSION['usr_alert'])) {
```

Comment:
Why do we need a session variable for this? Why not just have validateGpxFile() return the necessary status? In fact, why not have validateGpxFile() actually display the alert?

Reply:
The reverseGpx script is invoked as a form. When the form is submitted, the admintools page is exited, proceeding to the reverseGpx page. There is no html or javascript on that page. The idea is to take the admin back to the page where the form was submitted, with refreshed data. In order to get the error message to the admintools page, it has to be in a location where can be accessed by that page when it returns. Session variables allow temporary storage across pages. Secondarily, the point of the exercise was to use the newly defined functions to avoid duplicating code. validateGpxFile is returning the status - in this case to reverseGpx, in other cases to saveTab1 - via the session variable.

Reply:
Thanks for explaining the redirect issue. It just seems so hopelessly convoluted to me. I'm still thinking about this one.